tar/siginfo.c

```diff
@@ -115,6 +114,7 @@ void
 siginfo_setinfo(struct bsdtar *bsdtar, const char * oper, const char * path,
     int64_t size, int file_count, uint64_t total_uncompressed)
 {
+	static uint64_t lastprogress = 0;
```

Comment:
I used `static uint64_t lastprogress` because that's what happened in `raisesigs()` from `tar/storage/storage_write.c` (with `lastcheckpoint`).  I thought it was a nice solution, actually -- it keeps the variable scope isolated to the code that actually needs it.  Are you sure that you'd rather have it inside `struct siginfo_data`?

Reply:
It's nice for scoping reasons; but it's bad because aside from scope it's a global variable.  Singletons should be avoided except when there's a very good reason for them...